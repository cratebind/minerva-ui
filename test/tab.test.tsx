import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
  Tabs,
  TabList,
  Tab,
  TabPanels,
  TabPanel,
  ThemeProvider,
  defaultTheme,
} from '../src';

const ExampleTabs = () => {
  const selectedStyled = {
    bg: '#e0e0e0',
    borderColor: '#e0e0e0',
    borderRadius: '5px',
  };
  return (
    <Tabs>
      <TabList data-testid="TabList" bg="#f5f6fa" mb="20px">
        <Tab
          data-testid="active-tab"
          _selected={{ ...selectedStyled }}
          mr="15px"
        >
          My Account
        </Tab>
        <Tab _selected={{ ...selectedStyled }} mr="15px">
          Favorites
        </Tab>
        <Tab _selected={{ ...selectedStyled }} mr="15px">
          Orders
        </Tab>
        <Tab _selected={{ ...selectedStyled }}>Billing</Tab>
      </TabList>
      <TabPanels data-testid="TabPanels">
        <TabPanel data-testid="TabPanel">
          <p>My Account!</p>
        </TabPanel>
        <TabPanel>
          <p>Favorites!</p>
        </TabPanel>
        <TabPanel>
          <p>Orders!</p>
        </TabPanel>
        <TabPanel>
          <p>Billing!</p>
        </TabPanel>
      </TabPanels>
    </Tabs>
  );
};

describe('<Tabs />', () => {
  it('should change TabList background color', () => {
    const { container, getByTestId } = render(
      <ThemeProvider>
        <ExampleTabs />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    const tabList = getByTestId('TabList');
    expect(tabList).toHaveStyleRule('background-color', '#f5f6fa');
  });

  it('should change active tab styles', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <ExampleTabs />
      </ThemeProvider>
    );
    const activeTab = getByTestId('active-tab');
    expect(activeTab).toHaveStyleRule('background-color', '#e0e0e0', {
      modifier: '[aria-selected=true]',
    });
  });
});

['TabList', 'TabPanels', 'TabPanel'].forEach(componentName => {
  describe(`<${componentName} />`, () => {
    it('should be theme-able', () => {
      const backgroundColor = '#e9e9e9';
      const customTheme = {
        ...defaultTheme,
        [componentName]: {
          backgroundColor,
        },
      };
      const { container, getByTestId } = render(
        <ThemeProvider theme={customTheme}>
          <ExampleTabs />
        </ThemeProvider>
      );
      expect(container).toMatchSnapshot();
      const tabList = getByTestId(componentName);
      expect(tabList).toHaveStyleRule('background-color', backgroundColor);
    });
  });
});

// describe('<TabList />', () => {
//   it('should be theme-able', () => {
//     const backgroundColor = '#e9e9e9';
//     const customTheme = {
//       ...defaultTheme,
//       TabList: {
//         backgroundColor,
//       },
//     };
//     const { container, getByTestId } = render(
//       <ThemeProvider theme={customTheme}>
//         <ExampleTabs />
//       </ThemeProvider>
//     );
//     expect(container).toMatchSnapshot();
//     const tabList = getByTestId('TabList');
//     expect(tabList).toHaveStyleRule('background-color', backgroundColor);
//   });
// });

// describe('<TabPanels />', () => {
//   it('should be theme-able', () => {
//     const backgroundColor = '#e9e9e9';
//     const customTheme = {
//       ...defaultTheme,
//       TabList: {
//         backgroundColor,
//       },
//     };
//     const { container, getByTestId } = render(
//       <ThemeProvider theme={customTheme}>
//         <ExampleTabs />
//       </ThemeProvider>
//     );
//     expect(container).toMatchSnapshot();
//     const tabList = getByTestId('TabList');
//     expect(tabList).toHaveStyleRule('background-color', backgroundColor);
//   });
// });
