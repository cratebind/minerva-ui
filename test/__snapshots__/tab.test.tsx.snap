// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabList /> should be theme-able 1`] = `
.c0 {
  box-sizing: border-box;
  min-width: 0;
}

.c1 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #e9e9e9;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c3 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
  margin-right: 15px;
}

.c5 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
}

.c4:focus {
  color: #1a56db;
  outline: 0;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  border-width: 2px;
}

.c4[aria-selected=true],
.c4[data-selected=true],
.c4[data-selected] {
  background-color: #e0e0e0;
  border-color: #e0e0e0;
  border-radius: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #e9e9e9;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

<div>
  <div
    class="c0"
    data-orientation="horizontal"
    data-reach-tabs=""
  >
    <div
      aria-orientation="horizontal"
      class="c1 c2"
      data-reach-tab-list=""
      data-testid="TabList"
      role="tablist"
    >
      <button
        aria-controls="tabs--10--panel--0"
        aria-selected="true"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        data-selected=""
        data-testid="active-tab"
        id="tabs--10--tab--0"
        role="tab"
        tabindex="0"
        type="button"
      >
        My Account
      </button>
      <button
        aria-controls="tabs--10--panel--1"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--10--tab--1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Favorites
      </button>
      <button
        aria-controls="tabs--10--panel--2"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--10--tab--2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Orders
      </button>
      <button
        aria-controls="tabs--10--panel--3"
        aria-selected="false"
        class="c5 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--10--tab--3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Billing
      </button>
    </div>
    <div
      class="c0"
      data-reach-tab-panels=""
      data-testid="TabPanels"
    >
      <div
        aria-labelledby="tabs--10--tab--0"
        class="c0"
        data-reach-tab-panel=""
        data-testid="TabPanel"
        id="tabs--10--panel--0"
        role="tabpanel"
        tabindex="0"
      >
        <p>
          My Account!
        </p>
      </div>
      <div
        aria-labelledby="tabs--10--tab--1"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--10--panel--1"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Favorites!
        </p>
      </div>
      <div
        aria-labelledby="tabs--10--tab--2"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--10--panel--2"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Orders!
        </p>
      </div>
      <div
        aria-labelledby="tabs--10--tab--3"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--10--panel--3"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Billing!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TabPanel /> should be theme-able 1`] = `
.c0 {
  box-sizing: border-box;
  min-width: 0;
}

.c1 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c3 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
  margin-right: 15px;
}

.c5 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
}

.c6 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #e9e9e9;
}

.c4:focus {
  color: #1a56db;
  outline: 0;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  border-width: 2px;
}

.c4[aria-selected=true],
.c4[data-selected=true],
.c4[data-selected] {
  background-color: #e0e0e0;
  border-color: #e0e0e0;
  border-radius: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

<div>
  <div
    class="c0"
    data-orientation="horizontal"
    data-reach-tabs=""
  >
    <div
      aria-orientation="horizontal"
      class="c1 c2"
      data-reach-tab-list=""
      data-testid="TabList"
      role="tablist"
    >
      <button
        aria-controls="tabs--18--panel--0"
        aria-selected="true"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        data-selected=""
        data-testid="active-tab"
        id="tabs--18--tab--0"
        role="tab"
        tabindex="0"
        type="button"
      >
        My Account
      </button>
      <button
        aria-controls="tabs--18--panel--1"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--18--tab--1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Favorites
      </button>
      <button
        aria-controls="tabs--18--panel--2"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--18--tab--2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Orders
      </button>
      <button
        aria-controls="tabs--18--panel--3"
        aria-selected="false"
        class="c5 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--18--tab--3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Billing
      </button>
    </div>
    <div
      class="c0"
      data-reach-tab-panels=""
      data-testid="TabPanels"
    >
      <div
        aria-labelledby="tabs--18--tab--0"
        class="c6"
        data-reach-tab-panel=""
        data-testid="TabPanel"
        id="tabs--18--panel--0"
        role="tabpanel"
        tabindex="0"
      >
        <p>
          My Account!
        </p>
      </div>
      <div
        aria-labelledby="tabs--18--tab--1"
        class="c6"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--18--panel--1"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Favorites!
        </p>
      </div>
      <div
        aria-labelledby="tabs--18--tab--2"
        class="c6"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--18--panel--2"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Orders!
        </p>
      </div>
      <div
        aria-labelledby="tabs--18--tab--3"
        class="c6"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--18--panel--3"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Billing!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TabPanels /> should be theme-able 1`] = `
.c0 {
  box-sizing: border-box;
  min-width: 0;
}

.c1 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c3 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
  margin-right: 15px;
}

.c5 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
}

.c6 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #e9e9e9;
}

.c4:focus {
  color: #1a56db;
  outline: 0;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  border-width: 2px;
}

.c4[aria-selected=true],
.c4[data-selected=true],
.c4[data-selected] {
  background-color: #e0e0e0;
  border-color: #e0e0e0;
  border-radius: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

<div>
  <div
    class="c0"
    data-orientation="horizontal"
    data-reach-tabs=""
  >
    <div
      aria-orientation="horizontal"
      class="c1 c2"
      data-reach-tab-list=""
      data-testid="TabList"
      role="tablist"
    >
      <button
        aria-controls="tabs--14--panel--0"
        aria-selected="true"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        data-selected=""
        data-testid="active-tab"
        id="tabs--14--tab--0"
        role="tab"
        tabindex="0"
        type="button"
      >
        My Account
      </button>
      <button
        aria-controls="tabs--14--panel--1"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--14--tab--1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Favorites
      </button>
      <button
        aria-controls="tabs--14--panel--2"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--14--tab--2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Orders
      </button>
      <button
        aria-controls="tabs--14--panel--3"
        aria-selected="false"
        class="c5 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--14--tab--3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Billing
      </button>
    </div>
    <div
      class="c6"
      data-reach-tab-panels=""
      data-testid="TabPanels"
    >
      <div
        aria-labelledby="tabs--14--tab--0"
        class="c0"
        data-reach-tab-panel=""
        data-testid="TabPanel"
        id="tabs--14--panel--0"
        role="tabpanel"
        tabindex="0"
      >
        <p>
          My Account!
        </p>
      </div>
      <div
        aria-labelledby="tabs--14--tab--1"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--14--panel--1"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Favorites!
        </p>
      </div>
      <div
        aria-labelledby="tabs--14--tab--2"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--14--panel--2"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Orders!
        </p>
      </div>
      <div
        aria-labelledby="tabs--14--tab--3"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--14--panel--3"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Billing!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs /> should change TabList background color 1`] = `
.c0 {
  box-sizing: border-box;
  min-width: 0;
}

.c1 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c3 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
  margin-right: 15px;
}

.c5 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  font-weight: 500;
  border-width: 2px;
  margin-bottom: -2px;
  border: 0;
  font-size: 16px;
  background-color: transparent;
  border-bottom: 2px solid transparent;
}

.c4:focus {
  color: #1a56db;
  outline: 0;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  border-width: 2px;
}

.c4[aria-selected=true],
.c4[data-selected=true],
.c4[data-selected] {
  background-color: #e0e0e0;
  border-color: #e0e0e0;
  border-radius: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #f5f6fa;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

<div>
  <div
    class="c0"
    data-orientation="horizontal"
    data-reach-tabs=""
  >
    <div
      aria-orientation="horizontal"
      class="c1 c2"
      data-reach-tab-list=""
      data-testid="TabList"
      role="tablist"
    >
      <button
        aria-controls="tabs--1--panel--0"
        aria-selected="true"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        data-selected=""
        data-testid="active-tab"
        id="tabs--1--tab--0"
        role="tab"
        tabindex="0"
        type="button"
      >
        My Account
      </button>
      <button
        aria-controls="tabs--1--panel--1"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--1--tab--1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Favorites
      </button>
      <button
        aria-controls="tabs--1--panel--2"
        aria-selected="false"
        class="c3 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--1--tab--2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Orders
      </button>
      <button
        aria-controls="tabs--1--panel--3"
        aria-selected="false"
        class="c5 c4"
        data-orientation="horizontal"
        data-reach-tab=""
        id="tabs--1--tab--3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Billing
      </button>
    </div>
    <div
      class="c0"
      data-reach-tab-panels=""
      data-testid="TabPanels"
    >
      <div
        aria-labelledby="tabs--1--tab--0"
        class="c0"
        data-reach-tab-panel=""
        data-testid="TabPanel"
        id="tabs--1--panel--0"
        role="tabpanel"
        tabindex="0"
      >
        <p>
          My Account!
        </p>
      </div>
      <div
        aria-labelledby="tabs--1--tab--1"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--1--panel--1"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Favorites!
        </p>
      </div>
      <div
        aria-labelledby="tabs--1--tab--2"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--1--panel--2"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Orders!
        </p>
      </div>
      <div
        aria-labelledby="tabs--1--tab--3"
        class="c0"
        data-reach-tab-panel=""
        hidden=""
        id="tabs--1--panel--3"
        role="tabpanel"
        tabindex="-1"
      >
        <p>
          Billing!
        </p>
      </div>
    </div>
  </div>
</div>
`;
